app/services/course/assessment/question/programming/python/python_package_service.rb

```diff
@@ -198,6 +174,31 @@ def generate_zip_file(data_files_to_keep)
     tmp
   end
 
+  def zip_test_files(test_type, zip)
+    tests = @test_params[:test_cases]
+    index = 1
+
+    tests[test_type].try(:each) do |test|
```

Comment:
What you can do is to use `tests[test_type]&.each_with_index(1) do |test, index|`

`&.` is the safe navigation operator, similar to `try(:each)`. Also, the each_with_index loops through each element and also manages an index. 

The other way is to use `tests[test_type].try(:each_with_index, 1) do |test, index|` but I'd prefer the first one, more readable.